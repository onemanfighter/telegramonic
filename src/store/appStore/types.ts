import { StateCreator } from 'zustand';
import {
  AlertStateSlice,
  DrawerStateSlice,
  ModalStateSlice,
  ThemeStateSlice,
} from '../slice';

export interface AppStoreState {
  Theme: ThemeStateSlice;
  Alert: AlertStateSlice;
  Modal: ModalStateSlice;
  Drawer: DrawerStateSlice;
}

export type AppStoreSlice<T> = StateCreator<
  AppStoreState,
  [['zustand/immer', never]],
  [],
  T
>;
