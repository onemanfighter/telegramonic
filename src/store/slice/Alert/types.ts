export type AlertComponentState = {
  title: string;
  description: string;
  showIcon?: boolean;
  status?: 'error' | 'success' | 'warning' | 'info' | 'loading' | undefined;
  variant?: 'solid' | 'subtle' | 'left-accent' | 'top-accent';
};

export type AlertState = {
  alertVisibleState: boolean;
  alert: AlertComponentState;
};

export interface AlertStateAction {
  setAlertVisible: (visible: boolean) => void;
  setAlertStateData: (data: AlertComponentState) => void;
}

export type AlertStateSlice = AlertState & AlertStateAction;
