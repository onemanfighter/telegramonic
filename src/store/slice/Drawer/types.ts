export type DrawerData = {
  isOpen: boolean;
  title: string;
  children: React.ReactNode;
  footer: React.ReactNode;
  onModalClose: () => void;
};

export type DrawerState = {
  drawerData: DrawerData;
  openState: DrawerOpenState;
};

export enum DrawerOpenState {
  OPEN = 'OPEN',
  CLOSE = 'CLOSE',
}

export interface DrawerStateAction {
  openDrawer: (drawerData: DrawerData) => void;
  closeDrawer: () => void;
}

export type DrawerStateSlice = DrawerState & DrawerStateAction;
