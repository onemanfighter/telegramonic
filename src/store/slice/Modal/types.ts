export type ModalData = {
  title: string;
  message: string;
  onModalClose?: () => void;
  primaryButtonLabel?: string;
  onPrimaryButtonClick?: () => void;
};

export type ModalState = {
  modalData: ModalData;
  openState: ModelOpenState;
};

export enum ModelOpenState {
  OPEN = 'OPEN',
  CLOSE = 'CLOSE',
}

export interface ModalStateAction {
  openModal: (modalData: ModalData) => void;
  closeModal: () => void;
}

export type ModalStateSlice = ModalState & ModalStateAction;
