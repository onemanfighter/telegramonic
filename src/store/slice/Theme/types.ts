export enum Theme {
  LIGHT = 'light',
  DARK = 'dark',
}

export type ThemeState = {
  themeValue: Theme;
};

export interface ThemeStateActions {
  setTheme: (theme: Theme) => void;
}

export type ThemeStateSlice = ThemeState & ThemeStateActions;
