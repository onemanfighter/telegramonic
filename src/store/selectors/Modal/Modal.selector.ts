import { AppStoreState } from '@store';

const modalSelector = (state: AppStoreState) => ({
  openState: state.Modal.openState,
  modalData: state.Modal.modalData,
  openModal: state.Modal.openModal,
  closeModal: state.Modal.closeModal,
});

export { modalSelector };
