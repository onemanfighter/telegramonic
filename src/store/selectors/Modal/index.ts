export { modalSelector } from './Modal.selector';
