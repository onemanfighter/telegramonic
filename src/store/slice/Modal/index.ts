export { default as createModalSlice } from './Modal.slice';
export type {
  ModalState,
  ModalState as ModalProps,
  ModalStateSlice,
} from './types';
export { ModelOpenState } from './types';
