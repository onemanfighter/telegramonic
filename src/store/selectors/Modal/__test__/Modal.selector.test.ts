import { AlertComponentState, DrawerState, appStore } from '@store';
import { act, renderHook } from '@testing-library/react';
import { modalSelector } from '../Modal.selector';
import { ModalData } from 'src/store/slice/Modal/types';

jest.useFakeTimers();

describe('Modal selector', () => {
  const onModalCloseMock = jest.fn();
  const onPrimaryButtonClickMock = jest.fn();

  it('should return modal selector state and actions', () => {
    const modal = renderHook(() => appStore(modalSelector)).result.current;

    expect(modal).toMatchSnapshot();
  });

  it('should return modal state with initial value', () => {
    const { result } = renderHook(() => appStore(modalSelector));

    expect(result.current.modalData).toMatchSnapshot();
  });

  it('should return modal state on modal data set using openModal', () => {
    const modalData: ModalData = {
      title: 'title',
      message: 'message',
      onModalClose: onModalCloseMock,
      primaryButtonLabel: 'primaryButtonLabel',
      onPrimaryButtonClick: onPrimaryButtonClickMock,
    };
    const { result } = renderHook(() => appStore(modalSelector));

    act(() => {
      result.current.openModal(modalData);
    });

    expect(result.current.modalData).toMatchSnapshot();
  });

  it('should reset and close modal state on modal data set using closeModal', () => {
    const modalData: ModalData = {
      title: 'title',
      message: 'message',
      onModalClose: onModalCloseMock,
      primaryButtonLabel: 'primaryButtonLabel',
      onPrimaryButtonClick: onPrimaryButtonClickMock,
    };
    const { result } = renderHook(() => appStore(modalSelector));

    act(() => {
      result.current.openModal(modalData);
    });
    expect(result.current.modalData.title).toEqual('title');
    expect(result.current.modalData.message).toEqual('message');

    act(() => {
      result.current.closeModal();
      jest.runAllTimers();
    });

    expect(result.current.modalData.title).toEqual('');
    expect(result.current.modalData.message).toEqual('');
  });
});
