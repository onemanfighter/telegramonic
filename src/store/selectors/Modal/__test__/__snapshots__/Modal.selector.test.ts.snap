// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal selector should return modal selector state and actions 1`] = `
{
  "closeModal": [Function],
  "modalData": {
    "message": "",
    "onModalClose": [Function],
    "onPrimaryButtonClick": [Function],
    "primaryButtonLabel": "",
    "title": "",
  },
  "openModal": [Function],
  "openState": "CLOSE",
}
`;

exports[`Modal selector should return modal state on modal data set using openModal 1`] = `
{
  "message": "message",
  "onModalClose": [MockFunction],
  "onPrimaryButtonClick": [MockFunction],
  "primaryButtonLabel": "primaryButtonLabel",
  "title": "title",
}
`;

exports[`Modal selector should return modal state with initial value 1`] = `
{
  "message": "",
  "onModalClose": [Function],
  "onPrimaryButtonClick": [Function],
  "primaryButtonLabel": "",
  "title": "",
}
`;
